app/components/Admin/Room/RoomDetail.jsx

```diff
@@ -1,20 +1,90 @@
 "use strict"
+import R from 'ramda';
 import dom from '../../../utils/dom';
 import Room from './room-class.js'
+import Field from '../../Form/Field-class';
 
 var tempRoom = new Room();
 
-
-function saveRoom() {
+const {is, isEmpty, propEq, prop, all, filter, map, join, compose} = R;
+
+// Utils
+const allValid = all(prop('isValid'))
```

Comment:
Added some utils for getting validation messages out (it's for an alert if form is submitted that says which fields are invalid... so maybe we won't even use this part)